### DISCUSSION TITLE
Error: Got unexpected extra argument (app.main:app)

### DISCUSSION THREAD

**Discussion by vagarwal77** [tiangolo/uvicorn-gunicorn-fastapi-docker#323] [Questions] [answered]
vivekagarwal@-MacBook-Pro rwe_external_integrations % docker run -t exter-rwe:latest
Usage: uvicorn [OPTIONS] APP
Try 'uvicorn --help' for help.

Error: Got unexpected extra argument (app.main:app)
######
FROM tiangolo/uvicorn-gunicorn-fastapi:python3.8
COPY requirements.txt .
RUN pip install -r requirements.txt
COPY ./app /app
EXPOSE 6565
ENTRYPOINT ["uvicorn"]
CMD ["uvicorn", "app.main:app", "--host", "0.0.0.0", "--port", "6565"]

**Comment by tarnopolskyi** [comment]
Hey
seems like it's because of CMD definition. Don't know what exactly, but smth inside uvicorn can't start with args which uvicorn get from docker CMD 
try: 
CMD uvicorn  app.main:app --host 0.0.0.0 --port 6565

**Comment by liyuling** [comment]
Any update on this issue?

**Comment by osuzdalev** [comment]
> Hey seems like it's because of CMD definition. Don't know what exactly, but smth inside uvicorn can't start with args which uvicorn get from docker CMD try: CMD uvicorn app.main:app --host 0.0.0.0 --port 6565

This worked for me.

**Comment by muelletm** [comment]
This should also work:

```Dockerfile
ENTRYPOINT ["uvicorn"]
CMD ["app.main:app", "--host", "0.0.0.0", "--port", "6565"]
```

(otherwise you will execute `uvicorn uvicorn app.main:app ...`)

**Comment by tiangolo** [comment, marked as answer]
This Docker image expected you **not** to define any `ENTRYPOINT` nor `CMD`. :nerd_face: 

---

Now that Uvicorn supports managing workers with `--workers`, including restarting dead ones, there's no need for Gunicorn. That also means that it's much simpler to build a Docker image from scratch now, I updated the docs to explain it.

Because of that, I deprecated this Docker image: https://github.com/tiangolo/uvicorn-gunicorn-fastapi-docker#-warning-you-probably-dont-need-this-docker-image
